xonsh/proc.py

```diff
@@ -541,14 +542,38 @@ def _wcode_to_popen(code):
                                    "stdin_redirect",
                                    "stdout_redirect",
                                    "stderr_redirect",
-                                   "timestamp"])
+                                   "timestamp",
+                                   "executed_cmd"])
 
 class CompletedCommand(_CCTuple):
     """Represents a completed subprocess-mode command."""
 
     def __bool__(self):
         return self.returncode == 0
 
+    def __iter__(self):
+        stdout = self.stdout
+        start = 0;
+        end = 0
+        while end != -1:
+            end = stdout.find('\n', start)
+            if end == -1:                # no newlines, but possibly more text
+                snippet = stdout[start:]
+            elif stdout[end-1] == '\r':  # newline, check for CR
+                snippet = stdout[start:end-1]
+            else:                        # newline, no CR
+                snippet = stdout[start:end]
+            if snippet or not (end == -1):
```

Comment:
..a few things, if I might:

This loops indefinitely -- stdout.partition should be be pre.partition, yes?

And, as you said, this probably doesn't matter, but -- rstrip is greedy.  If someone had:
"\r\r\r\n", the string yielded should, in theory, be "\r\r".  ..that's why I did the check for and removal of just one char.  I know it's finicky beyond relevance, as it's not likely anyone needs to retain extra \r's.  It just bugs a part of my brain.  OK if I satisfy it?

Lastly, and more likely to be a usefully valid point -- the code I had doesn't yield anything if there's no text (including no \n's).  To me, that seems the right way to do it, as it distinguishes between no output and a single newline.  I'm looking at, as the result:

```python
pre = self.stdout
post = None

if not pre:
    raise StopIteration()

while post != '':
    pre, sep, post = pre.partition('\n')
    pre = pre[:-1] if pre[-1] == '\r' else pre    # this line may be optional since we use universal newlines.
    yield pre
    pre = post
```

..you're right, partition is way more straightforward than what I had.  I'll have to tuck that away and use it.  :-)